File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/networking/socket/WalletTopUp.kt
package com.innomalist.taxi.common.networking.socket

import com.innomalist.taxi.common.networking.socket.interfaces.SocketRequest
import com.innomalist.taxi.common.utils.Adapters
import org.json.JSONObject

class WalletTopUp(gatewayId: Int, currency: String, token: String, amount: Double, pin: Int? = null, otp: Int? = null, transactionId: String? = null) : SocketRequest() {
    init {
        val dto = TopUpDto(gatewayId, amount, currency, token, pin, otp, transactionId)
        val obj = JSONObject(Adapters.moshi.adapter<TopUpDto>(TopUpDto::class.java).toJson(dto))
        this.params = arrayOf(obj)
    }
}

data class TopUpDto (
    val gatewayId: Int,
    val amount: Double,
    val currency: String,
    val token: String,
    val pin: Int? = null,
    val otp: Int? = null,
    val transactionId: String? = null
)

File: Ridy-taxi-app/ios/common/Networking/WebSocket/WalletTopUp.swift
//
//  WalletTopUp.swift
//  Shared
//
//  Created by Manly Man on 11/23/19.
//  Copyright © 2019 Innomalist. All rights reserved.
//

import Foundation

public final class WalletTopUp: SocketRequest {
    public typealias ResponseType = EmptyClass
    public var params: [Any]?
    
    public init(dto: WalletTopUpDTO) {
        self.params = [try! dto.asDictionary()]
    }
    
    required public init() {}
}

public struct WalletTopUpDTO: Codable {
    var gatewayId: Int
    var amount: Double
    var currency: String
    var token: String
    var pin: Int?
    var otp: Int?
    var transactionId: String?
}
